twilio/endpoint.bal

```diff
@@ -36,14 +36,24 @@ public isolated client class Client {
     # + twilioConfig - Twilio connection configuration 
     # + return - `http:Error` in case of failure to initialize or `null` if successfully initialized
     public isolated function init(ConnectionConfig twilioConfig, http:ClientConfiguration httpClientConfig = {}) returns error? {
-        self.accountSId = twilioConfig.accountSId;
-        auth:CredentialsConfig authConfig = {
-            username: twilioConfig.accountSId,
-            password: twilioConfig.authToken
-        };
-        http:ClientConfiguration httpConfig = httpClientConfig;
-        httpConfig.auth = authConfig;
-        self.basicClient = check new (TWILIO_API_BASE_URL, httpConfig);
+        self.accountSId = twilioConfig.authConfig.accountSId;
+        if (twilioConfig.authConfig is TokenBasedAuthentication) {
+            auth:CredentialsConfig authConfig = {
+                username: twilioConfig.authConfig.accountSId,
+                password: twilioConfig.authConfig?.authToken.toString()
+            };
+            http:ClientConfiguration httpConfig = httpClientConfig;
+            httpConfig.auth = authConfig;
+            self.basicClient = check new (TWILIO_API_BASE_URL, httpConfig);
```

Comment:
You can take the following code segment out of the if else block to avoid code repetition

```
http:ClientConfiguration httpConfig = httpClientConfig;
httpConfig.auth = authConfig;
self.basicClient = check new (TWILIO_API_BASE_URL, httpConfig);
```